src/controller/CHIPDeviceController.cpp

```diff
@@ -838,18 +857,31 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, Re
                 ExitNow(CHIP_ERROR_INTERNAL);
             }
             mRendezvousParametersForDeviceDiscoveredOverWiFiPAF = params;
-            auto nodeId                                         = params.GetPeerAddress().GetRemoteId();
             const SetupDiscriminator connDiscriminator(params.GetSetupDiscriminator().value());
             VerifyOrReturnValue(!connDiscriminator.IsShortDiscriminator(), CHIP_ERROR_INVALID_ARGUMENT,
                                 ChipLogError(Controller, "Error, Long discriminator is required"));
             uint16_t discriminator              = connDiscriminator.GetLongValue();
             WiFiPAF::WiFiPAFSession sessionInfo = { .role          = WiFiPAF::WiFiPafRole::kWiFiPafRole_Subscriber,
                                                     .nodeId        = nodeId,
                                                     .discriminator = discriminator };
-            ReturnErrorOnFailure(
-                DeviceLayer::ConnectivityMgr().GetWiFiPAF()->AddPafSession(WiFiPAF::PafInfoAccess::kAccNodeInfo, sessionInfo));
-            DeviceLayer::ConnectivityMgr().WiFiPAFSubscribe(discriminator, reinterpret_cast<void *>(this),
-                                                            OnWiFiPAFSubscribeComplete, OnWiFiPAFSubscribeError);
+            ReturnErrorOnFailure(pafLayer.AddPafSession(WiFiPAF::PafInfoAccess::kAccNodeInfo, sessionInfo));
+            err = DeviceLayer::ConnectivityMgr().WiFiPAFSubscribe(discriminator, reinterpret_cast<void *>(this),
+                                                                  OnWiFiPAFSubscribeComplete, OnWiFiPAFSubscribeError);
+            if (err != CHIP_NO_ERROR)
+            {
+                pafLayer.RmPafSession(WiFiPAF::PafInfoAccess::kAccNodeInfo, sessionInfo);
+                return err;
+            }
+            // Save the subscribe_id to mPafSessionIds[]
```

Comment:
Where do we remove things from mPafSessionIds?  We don't seem to do that anywhere....

But more importantly: what's the point of mPafSessionIds?  We never read it, only write it... https://github.com/project-chip/connectedhomeip/pull/38321#discussion_r2068609479 claims "This should be cleaned up (or released) after commissioning completed", but nothing does that, that I can see.

In particular, DeviceCommissioner instances are _not_ shut down when commissioning is completed.  These are long-lived objects that are potentially managing the entire fabric.